Nodejs/Product/Nodejs/Options/NodejsNpmOptionsControl.cs

```diff
@@ -42,20 +36,26 @@ public partial class NodejsNpmOptionsControl : UserControl {
         }
 
         private void ClearCacheButton_Click(object sender, EventArgs e) {
+            bool didClearNpmCache = DeleteCacheDirectory("npm cache", NodejsConstants.NpmCachePath);
+            bool didClearTools = DeleteCacheDirectory("NTVS external tools", NodejsConstants.ExternalToolsPath);
+
+            _cacheClearedSuccessfully.Visible = (didClearNpmCache && didClearTools);
+        }
+
+        private static bool DeleteCacheDirectory(string displayName, string cachePath) {
             try {
-                Directory.Delete(_npmCachePath, true);                    
-                _cacheClearedSuccessfully.Visible = true;
+                Directory.Delete(cachePath, true);
+               return true;
             } catch (DirectoryNotFoundException) {
                 // Directory has already been deleted. Do nothing.
-                _cacheClearedSuccessfully.Visible = true;
+                return true;
             } catch (IOException exception) {
                 // files are in use or path is too long
```

Comment:
On < Node 6, the max path was 273 :sad:

I changed the logic to use `robocopy` to nuke the directory instead. This was tested with Node 6 and Node 0.12, and both seem to work fine now.

For reference, another way to delete long path files is with the [Windows `DeleteFile` function](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363915%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396):

> In the ANSI version of this function, the name is limited to MAX_PATH characters. To extend this limit to 32,767 wide characters, call the Unicode version of the function and prepend "\\\\?\" to the path.